dask/array/chunk.py

```diff
@@ -176,16 +176,50 @@ def trim(x, axes=None):
     broadcast_to = npcompat.broadcast_to
 
 
-def topk(k, x):
-    """ Top k elements of an array
-
-    >>> topk(2, np.array([5, 1, 3, 6]))
-    array([6, 5])
+def topk(a, k, axis, keepdims):
+    """Kernel of topk and argtopk.
+    Extract the k largest elements from a on the given axis.
+    If k is negative, extract the -k smallest elements instead.
+    Note that, unlike in the parent function, the returned elements
+    are not sorted internally.
+    """
+    axis = axis[0]
+    if abs(k) >= a.shape[axis]:
+        return a
+    a = np.partition(a, -k, axis=axis)
+    # a = a[-k:] if k>0 else a[:-k], on arbitrary axis
+    a = a[[
+        (slice(-k, None) if k > 0 else slice(None, -k))
+        if i == axis else slice(None)
+        for i in range(a.ndim)
+    ]]
+    return a
+
+
+def topk_postproc(a, k, axis):
```

Comment:
These names show up in the graphs, and are only marginally shorter than the non-abbreviated versions. I'd prefer `topk_postprocess` and `argtopk_preprocess`.